### PR TITLE
Enable distributed 3D simulation option

### PR DESCRIPTION
**Pull request by akshaysridhar**
Modify function spaces to enable distributed simulations
cc @sriharshakandala 

- [x] Distributed 3D Box  
- [x] Distributed 3D Sphere
- [ ] Replicate ClimaCore tests with modifications

### CODE REVIEW

**Comment by akshaysridhar** [comment]
Awaiting changes from ClimaCore #631 before merging this (it should simplify ClimaAtmos changes a fair bit)

File: src/Simulations/simulation.jl
Comment (bischtob) [inline src/Simulations/simulation.jl:46]: `comms_ctx_xyz` could be `comms_context_xyz`

File: test/test_domains.jl
Comment (bischtob) [inline test/test_domains.jl:29]: add a todo for the upcoming tests

**Review by bischtob — CHANGES_REQUESTED** [review]
Nice. Just a few tiny things.

File: src/Models/Nonhydrostatic3DModels/equations_base_model.jl
Comment (sriharshakandala) [inline src/Models/Nonhydrostatic3DModels/equations_base_model.jl:139]: We are currently using a conservatively sized comms_ctx for both face and center DSS in ClimaCore

**Comment by akshaysridhar** [comment]
631 in ClimaCore is in, I'll udpate this PR  and request a second review.

**Pull request closed without merging**
